cmd/edit/addon/cmd.go

```diff
@@ -0,0 +1,226 @@
+		}
+	}()
+
+	// Get the client for the OCM collection of clusters:
+	ocmClient := ocmConnection.ClustersMgmt().V1()
+
+	// Try to find the cluster:
+	reporter.Debugf("Loading cluster '%s'", clusterKey)
+	cluster, err := clusterprovider.GetCluster(ocmClient.Clusters(), clusterKey, awsCreator.ARN)
+	if err != nil {
+		reporter.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	if cluster.State() != cmv1.ClusterStateReady {
+		reporter.Errorf("Cluster '%s' is not yet ready", clusterKey)
+		os.Exit(1)
+	}
+
+	parameters, err := clusterprovider.GetAddOnParameters(ocmClient.Addons(), addOnID)
+	if err != nil {
+		reporter.Errorf("Failed to get add-on '%s' parameters: %v", addOnID, err)
+		os.Exit(1)
+	}
+
+	addOnInstallation, err := clusterprovider.GetAddOnInstallation(ocmClient.Clusters(),
+		clusterKey, awsCreator.ARN, addOnID)
+	if err != nil {
+		reporter.Errorf("Failed to get add-on '%s' installation: %v", addOnID, err)
+		os.Exit(1)
+	}
+
+	var params []clusterprovider.AddOnParam
+	if parameters.Len() > 0 {
+		parameters.Each(func(param *cmv1.AddOnParameter) bool {
+			// Find the installation parameter corresponding to the addon parameter
+			var addOnInstallationParam *cmv1.AddOnInstallationParameter
+			addOnInstallation.Parameters().Each(func(p *cmv1.AddOnInstallationParameter) bool {
+				if p.ID() == param.ID() {
+					addOnInstallationParam = p
+					return false
+				}
+				return true
+			})
+			if !param.Editable() {
+				return true
+			}
+			input := interactive.Input{
+				Question: param.Name(),
+				Help:     param.Description(),
+				Required: param.Required(),
+			}
+
+			var val string
+			switch param.ValueType() {
+			case "boolean":
+				var boolVal bool
+				input.Default, _ = strconv.ParseBool(addOnInstallationParam.Value())
+				boolVal, err = interactive.GetBool(input)
+				if boolVal {
+					val = "true"
+				} else {
+					val = "false"
+				}
+			case "cidr":
+				var cidrVal net.IPNet
+				if addOnInstallationParam.Value() != "" {
+					_, defaultIDR, _ := net.ParseCIDR(addOnInstallationParam.Value())
+					input.Default = *defaultIDR
+				}
+				cidrVal, err = interactive.GetIPNet(input)
+				val = cidrVal.String()
+			case "number":
+				var numVal int
+				input.Default, _ = strconv.ParseInt(addOnInstallationParam.Value(), 10, 0)
+				numVal, err = interactive.GetInt(input)
+				val = strconv.Itoa(numVal)
+			case "string":
+				input.Default = addOnInstallationParam.Value()
+				val, err = interactive.GetString(input)
+			}
+			if err != nil {
+				reporter.Errorf("Expected a valid value for '%s': %v", param.ID(), err)
+				os.Exit(1)
+			}
+
+			if val != "" {
+				val = strings.Trim(val, " ")
+			}
+			if val != "" && param.Validation() != "" {
+				isValid, err := regexp.MatchString(param.Validation(), val)
+				if err != nil || !isValid {
+					reporter.Errorf("Expected %v to match /%s/", val, param.Validation())
+					os.Exit(1)
+				}
```

Comment:
I assume you are doing this here to avoid an unnecessary API call? Should be fine as long as we never change how we process param.Validation

Reply:
Yes, we want to fail early if possible, to avoid unnecessarily leading the user through all the options if we already know it will fail